PyTDF/MapReduce.py

```diff
@@ -0,0 +1,135 @@
+class MapReduce(object):
+    """
+    Class for performing 
+    MapReduce on datasets
+
+    """
+    def __init__(self):
+        
+        self.actions = 0
+        self.tfs = 0
+        self.operations = []
+        self.action_node_map = {}
+
+    def execute_from_functions(self, mapper, reducer):
+        """
+        Use DistROOT to execute a distributed job
+        with the given mapper and reducer functions
+
+        TODO(shravan97)
+        """
+
+        pass
+
+    def _dfs(self, node, prev = 't', ops = []):
+        """
+        Method that does a depth-first
+        traversal and takes note of the 
+        necessary operations to carry out
+
+        """
+        
+        ops_copy = list(ops)
+
+        if not node.operation: # If root
+            for n in node.next_nodes:
+                self._dfs(n)
+            return
+
+        if node.operation.op_type == "a":
+            self.actions+=1
+            
+            ops_copy.append(node.operation)
```

Comment:
If there are only actions and transformations, you can move out of the if this append, since it is done in all the cases and sub-cases

Reply:
`ops_copy.append(node.operation)` has to be done inside the if/else statements because, they will get appended to the `operations` array, which additionally depends on `cur` and `prev` variables (they are just the current and previous names of actions/transformations)